### PR TITLE
Support version #s as strings, and remove some storages

### PR DESCRIPTION
**Pull request by EricWittmann**
This PR represents the following significant changes:

1. Remove the following storages:  **asyncmem**, **streams**, **infinispan**
2. Add storage support for treating versions as strings (optionally provided by end users)
3. Replace the in-memory storage with a H2 SQL storage

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/storage/impl/SimpleMapRegistryStorage.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 
+import org.apache.kafka.connect.errors.AlreadyExistsException;
```
Comment (famartinrh) [inline app/src/main/java/io/apicurio/registry/storage/impl/SimpleMapRegistryStorage.java:8]: this seems wrong

**Review by alesj — APPROVED** [review]

**Comment by famartinrh** [comment]
@EricWittmann you will need this https://github.com/EricWittmann/apicurio-registry/pull/31

File: app/src/main/java/io/apicurio/registry/storage/impl/SimpleMapRegistryStorage.java (hunk 1 above)
Comment (EricWittmann) [reply app/src/main/java/io/apicurio/registry/storage/impl/SimpleMapRegistryStorage.java:8]: It is wrong.  Good catch.  Fortunately this class is being deleted. :)

**Comment by famartinrh** [comment]
kubernetes tests are going to fail :)

File: app/src/main/java/io/apicurio/registry/storage/impl/sql/InMemoryRegistryStorage.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020 Red Hat
+ * Copyright 2020 JBoss Inc
```
Comment (famartinrh) [inline app/src/main/java/io/apicurio/registry/storage/impl/sql/InMemoryRegistryStorage.java:2]: :)

**Comment by EricWittmann** [comment]
@famartinrh I can't get all the tests to pass.  Can you have a look?

**Pull request merged**